respect-app-compose/src/commonMain/kotlin/world/respect/app/viewmodel/CurriculumEditViewModel.kt

```diff
@@ -0,0 +1,208 @@
+        viewModelScope.launch {
+            _uiState.value = _uiState.value.copy(isLoading = true, error = null)
+
+            try {
+                val currentState = _uiState.value
+                val curriculum = Curriculum(
+                    id = currentState.id,
+                    name = currentState.name,
+                    description = currentState.description,
+                    isActive = true
+                )
+
+                val result = saveCurriculumUseCase(curriculum)
+
+                result.fold(
+                    onSuccess = { savedCurriculum ->
+                        _uiState.value = _uiState.value.copy(isLoading = false)
+                        _navCommandFlow.emit(
+                            NavCommand.Navigate(
+                                CurriculumDetail(
+                                    curriculumId = savedCurriculum.id,
+                                    curriculumName = savedCurriculum.name
+                                )
+                            )
+                        )
+                    },
+                    onFailure = { exception ->
+                        val errorMessage = exception.message ?: SAVE_FAILED_ERROR
+                        _uiState.value = _uiState.value.copy(
+                            isLoading = false,
+                            error = errorMessage
+                        )
+                    }
+                )
+
+            } catch (e: Exception) {
+                val errorMessage = e.message ?: UNKNOWN_ERROR_MESSAGE
+                _uiState.value = _uiState.value.copy(
+                    isLoading = false,
+                    error = errorMessage
+                )
+            }
+        }
+    }
+    private fun loadCurriculum(id: String) {
+        viewModelScope.launch {
+            _uiState.value = _uiState.value.copy(isLoading = true, error = null)
+
+            try {
+                val curriculum = getCurriculumByIdUseCase(id)
+
+                if (curriculum != null) {
+                    _uiState.value = _uiState.value.copy(
+                        curriculum = curriculum,
+                        name = curriculum.name,
+                        id = curriculum.id,
+                        description = curriculum.description,
+                        isLoading = false
+                    )
+                } else {
+                    _uiState.value = _uiState.value.copy(
+                        isLoading = false,
+                        error = CURRICULUM_NOT_FOUND_ERROR
+                    )
+                }
+
+            } catch (e: Exception) {
+                val errorMessage = e.message ?: LOAD_FAILED_ERROR
+                _uiState.value = _uiState.value.copy(
+                    isLoading = false,
+                    error = errorMessage
+                )
+            }
+        }
+    }
+    companion object {
+        private const val VALIDATION_NAME_REQUIRED = "name_required"
```

Comment:
Strings to show the user should never be string literals; they should be localized and added in strings.xml